### PR TITLE
[AIRFLOW-6860] Default ignore_first_depends_on_past to True

### PR DESCRIPTION
**Pull request by pingzh**
Default `ignore_first_depends_on_past` is very confusing for users when doing backfill with `depends_on_past`. We should make it true.

---
Issue link: [AIRFLOW-6860](https://issues.apache.org/jira/browse/AIRFLOW-6860)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

@KevinYang21

### CODE REVIEW

**Review by KevinYang21 — COMMENTED** [review]
+1 for this new behavior but we need to include the UPDATING.md file

**Review by KevinYang21 — APPROVED** [review]

File: airflow/cli/commands/dag_command.py [hunk 1]
```diff
@@ -67,6 +67,13 @@ def dag_backfill(args, dag=None):
 
     signal.signal(signal.SIGTERM, sigint_handler)
 
+    import warnings
+    warnings.warn('--ignore_first_depends_on_past is deprecated as the value is always set to True',
```
Comment (zhongjiajie) [inline airflow/cli/commands/dag_command.py:71]: Should be  `--ignore-first-depends-on-past` due to https://github.com/apache/airflow/pull/7148

**Review by zhongjiajie — COMMENTED** [review]
Some minor suggest here

File: airflow/cli/commands/dag_command.py (hunk 1 above)
Comment (zhongjiajie) [reply airflow/cli/commands/dag_command.py:71]: will submit new PR to correct it.

Comment (pingzh) [reply airflow/cli/commands/dag_command.py:71]: thanks.

**Pull request merged**